Protecs/src/main/java/uk/ac/ucl/protecs/behaviours/Demography.java

```diff
@@ -0,0 +1,224 @@
+				break;
+			}
+			// ------------------------------------------------------------------------------------------------------------
+			case "female": {
+				myMortalityLikelihood = world.params.getLikelihoodByAge(
+						world.params.prob_death_by_age_female, world.params.all_cause_death_age_params, target.getAge());	
+				break;
+			}
+			// ------------------------------------------------------------------------------------------------------------
+			default: System.out.println("Sex/age based mortality not found");
+			}
+				
+			// check if this person is going to die in the next year
+			String nextStep = "noDeath";
+			
+			if (arg0.random.nextDouble() <= myMortalityLikelihood){
+				nextStep = "Death";
+				}
+			
+			switch (nextStep) {
+			// ------------------------------------------------------------------------------------------------------------
+			case "Death":{
+				// choose a day to die this year, then schedule this death to take place
+				this.tickToCauseMortality = arg0.random.nextInt(365) * world.params.ticks_per_day;
+				arg0.schedule.scheduleOnce(arg0.schedule.getTime() + this.tickToCauseMortality, this);
+				break;
+			}
+			// ------------------------------------------------------------------------------------------------------------
+			case "noDeath":{
+				arg0.schedule.scheduleOnce(arg0.schedule.getTime() + this.ticksUntilNextMortalityCheck, this);
+				break;
+			}
+			// ------------------------------------------------------------------------------------------------------------
+			default: {System.out.println("Mortality not determined");
+			}
+			}
+			
+		}
+		
+	}
+	public class Births implements Steppable{
+		
+		Person target;
+		int ticksUntilNextBirthCheck = 0;
+		int tickToCauseBirth = Integer.MAX_VALUE;
+		int daysToRescheduleNextBirth = Integer.MAX_VALUE;
+		WorldBankCovid19Sim world;
+		public Births( Person p, int ticksPerDay, WorldBankCovid19Sim myWorld ) {
+			this.target = p;
+			this.ticksUntilNextBirthCheck = 365 * ticksPerDay;
+			this.world = myWorld;
+		} 
+		@Override
+		public void step(SimState arg0) {
+			// If a due date has been created cause a birth on this day
+			if (this.tickToCauseBirth < Integer.MAX_VALUE) {
+				createBirth(arg0, target.isAlive());
+				postBirthRescheduling(arg0, target.isAlive());
+			}
+			else {
+				determineGivingBirth(arg0, target.isAlive());
+				
+			}
+			
+		}
+		private void postBirthRescheduling(SimState arg0, boolean isAlive) {
+			if (isAlive) {
+			// reset tickToCauseBirth so this pathway can be used again
+			this.tickToCauseBirth = Integer.MAX_VALUE;
+			// reschedule the check to occur next year
+			int currentTime = (int) (arg0.schedule.getTime() / world.params.ticks_per_day);
+			int currentYear = (int) Math.floor(currentTime / 365);
+			int nextYear = (currentYear + 1) * 365;
+			arg0.schedule.scheduleOnce(nextYear * world.params.ticks_per_day, this);
```

Comment:
is it necessarily the case that the person can't give birth for a year after giving birth?
Also - could we keep "ticks per year" as a pre-computed value and add it to the current time to get something that fires next year? That is, are we synchronising to the birth-giver's years, or to the calendar years?

Reply:
So this block is for people who have just given birth this year to check whether they will give birth again next year.

Say a person gave birth on day 340. From line 144 we have currentYear = floor(340/365) = 0. That means that at the start of the next year we will need to see if they will give birth again (So next year = (0 + 1) * 365 = 365. 

Finally the check whether this person will give birth in the following year happens on day 365 or year 1 of this sim.

Reply:
Obviously this is simplistic and not really that biologically realistic but I would say that unless we are doing more pregnancy/newborn specific health modelling it's probably good enough for now as at the population scale we're seeing appropriate yearly birth rates.

Reply:
Finally, ticks per year is a good idea!